### PR TITLE
Bug 1370594: New Homepage callouts, newsletter updates

### PR DESCRIPTION
**Pull request by stephaniehobson**
- Simplify callout code
- Move newsletter signup to callout
- Group newsletter text and form in wrapping containers
- Add new classes to newsletter for styling purposes
- Newsletter gets better tablet layout
- Newsletter goes to grid layout at small desktop screen sizes

Testing notes:
- if there are two newsletter signups on the page the JS fails
- newsletter signup on article and profile pages need testing too
- should I just remove the saucelabs code?

### CODE REVIEW

File: kuma/static/styles/components/home/column-callout.scss [hunk 1]
```diff
@@ -138,28 +198,68 @@ DevRel Survey promo
...
         }
     }
+
+    span {
+        padding: $grid-spacing;
+    }
+}
+
+/* larger teaser image gets hidden and shown at a different rate than the usual callouts */
+
```
Comment (schalkneethling) [inline kuma/static/styles/components/home/column-callout.scss]: Nit: empty line

File: kuma/static/styles/components/newsletter.scss [hunk 2]
```diff
@@ -136,10 +181,91 @@ input.newsletter-input-email {
...
+- it gets the .newsletter-box wrapper in articles
+-------------------------------------------------------------- */
+.newsletter-box {
+    position: relative;
+    border-top: 2px solid $text-color;
+    padding: $grid-spacing 0;
+    background-color: #fff;
+
+    .newsletter-teaser {
+
```
Comment (schalkneethling) [inline kuma/static/styles/components/newsletter.scss]: Nit: additional empty line?

**Comment by schalkneethling** [comment]
@stephaniehobson Is this expected?

screen shot 2017-07-12 at 15 28 31

I did `docker-compose down`, `docker-compose up -d` and `docker exec -i kuma_web_1 make build-static` but still see both newsletter forms.

> if there are two newsletter signups on the page the JS fails

Not sure if you were referring to the above scenario here. But yes, the field only expands to show the privacy notice for the first form.

**Comment by schalkneethling** [comment]
Between 770px to ~812px the button drops to just below the input field.

screen shot 2017-07-12 at 15 35 47

**Comment by schalkneethling** [comment]
Same happens for 481px to 530px

**Comment by schalkneethling** [comment]
Again not sure if this form should be present on the landing page but, at 769px the button extends beyond the input field's width:

screen shot 2017-07-12 at 15 40 46

**Comment by schalkneethling** [comment]
On a page such as en-US/docs/Learn the padding on the left is removed below 770px

screen shot 2017-07-12 at 15 44 49

**Comment by schalkneethling** [comment]
I also notice the same behaviour as in https://github.com/mozilla/kuma/pull/4310#issuecomment-314772522 on the profile edit page

**Comment by stephaniehobson** [comment]
With #4301 merged all the problems solved by the duplication should be fixed :)

**Comment by schalkneethling** [comment]
I still have the two newsletter forms on the landing page. I rebuilt all static assets with `docker exec -i kuma_web_1 make build-static` after pulling the latest from the branch.

**Comment by stephaniehobson** [comment]
I think you'll need to rebase on master as well.

**Comment by schalkneethling** [comment]
> I think you'll need to rebase on master as well. 
Gotcha, thought it was already merged with master 👍

**Comment by schalkneethling** [comment]
The various content sections here does not line up

screen shot 2017-07-14 at 10 33 36

**Comment by schalkneethling** [comment]
The colour contrast of the "Privacy Policy" link text is really hard to read and also fails WCAG. I reckon the same fix you made for the links in the masthead would work here.

screen shot 2017-07-14 at 10 33 36

**Comment by schalkneethling** [comment]
On success the "Thank you" text is displayed below the form, instead of replacing it.

screen shot 2017-07-14 at 10 44 54

Looks like it is this `grid` block that overrides the `.hidden` class.

```
.newsletter-box .newsletter-form {
        display: grid;
        grid-column-gap: 20px;
        grid-template-columns: 1fr 1fr; 
}
```

**Comment by schalkneethling** [comment]
Not sure if this is by design but, it feels to me that the form field do not line up horizontally with the copy on the left. Thoughts?

screen shot 2017-07-14 at 10 47 57

**Comment by schalkneethling** [comment]
The issues mentioned in https://github.com/mozilla/kuma/pull/4310#issuecomment-314771283 and https://github.com/mozilla/kuma/pull/4310#issuecomment-314771702 are still present. I am on the fence on this one though.

It is an inconsistency but, I would not consider it a blocker to merging this PR.

**Comment by schalkneethling** [comment]
When the layout changes at 760px, the "Get involved" callout drops down to just above the footer, as expected but, it's layout is broken.

screen shot 2017-07-14 at 11 10 09

**Comment by schalkneethling** [comment]
I still notice this problem: https://github.com/mozilla/kuma/pull/4310#issuecomment-314774164

**Comment by schalkneethling** [comment]
I also still see this behaviour on the profile edit page: https://github.com/mozilla/kuma/pull/4310#issuecomment-314772522

**Comment by stephaniehobson** [comment]
Okay, I think I've fixed the newsletter forms now except for the homepage between 1025 and 1034. This problem will also affect languages the same length and longer than English. I would like to launch with this small bug given how small our tablet usage is and the fact that it doesn't correspond to a common device size.

The intention for the home page block is that the text lines up with the top of the block and the inputs line up with the bottom (in line with the button in the promo block). The design direction was for the form fields to take up all available height perfectly but I'm out of time to try to get this working.

Other things fixed in most recent push:
- swapped out the survey promo for the sauce labs promo
- privacy policy link is now black with an underline
- the newsletter input field now extends to 100% width to match the submit button when appropriate
- submit button does not fall to line below between 481px to 530px  or 770px to ~812px 
- form is hidden when success message displays
- make home page newsletter form field and submit buttons bigger at desktop size 

The "Get Involved" block was broken by a different PR and will be addressed by a later PR.

**Comment by stephaniehobson** [comment]
@schalkneethling Updated 🌮

**Comment by schalkneethling** [comment]
Submit button arrow is of centre:

screen shot 2017-07-19 at 15 13 00

**Comment by schalkneethling** [comment]
A possible a11y issue happens at the following form layout:

screen shot 2017-07-19 at 15 17 20

The tab order changes from 
input -> checkbox -> Privacy Policy -> Submit 
to
input -> Submit -> checkbox -> Privacy Policy

**Comment by schalkneethling** [comment]
On the "Learn web development" page, there is an `x` icon to dismiss the form. Clicking the `x` does not remove the form. Once you refresh the page, the `x` icon is removed but, the form is still visible. I do not see any error reported in the console.

newsletter-form-dismiss

**Comment by schalkneethling** [comment]
Seems like Github does not support animated gifs, here is the described behaviour from above:

https://cl.ly/0M3I3W1y191k

**Comment by schalkneethling** [comment]
'newletterHide: true` is set in `localStorage` though:

screen shot 2017-07-19 at 15 32 13

**Comment by schalkneethling** [comment]
And so, clicking the `x` to dismiss the form works in IE ¯\_(ツ)_/¯

**Comment by schalkneethling** [comment]
Also, the a11y problem is only in Firefox

**Comment by schalkneethling** [comment]
Ok. Other than the items mentioned above. This is looking great! r+w/c 🍹

**Comment by stephaniehobson** [comment]
Fixed the newsletter not dismissing and centered the arrow. We're going to have to live with the weird tab order in Firefox for now :/

@schalkneethling thank you for your consistently high quality, through, and in-depth code reviews and functionality testing. 🍺

**Pull request merged**